### DISCUSSION TITLE
Admin API - add identity provider to the user?

### DISCUSSION THREAD

**Discussion by Ketec** [keycloak/keycloak#32017] [Q&A] [answered]
the REST API doc has:
`https://<authurl>/auth/admin/realms/<myrealm>/users/<user-is>/federated-identity/<idpalias>`
But there is no payload to specify userID/userName as the required parameters.
It mentions only "social providers".

i tried sending the data in the body, but it still gets 500:
`Cannot invoke "org.keycloak.representations.idm.FederatedIdentityRepresentation.getUserId()" because "rep" is null`

Is there no API to add other providers? I need to switch over 5000 users to a new IDP (in parallel) - and doing it in admin UI manually is not an option.

**Comment by danielFesenmeyer** [comment]
Which version of Keycloak are you using?

With Keycloak 24.0.6, it works fine - at least when using the Java Admin Client. We are using it in our tests.

Btw, the best way to figure out the required representation, is to look at the sources of the Java Admin Client: https://github.com/keycloak/keycloak/blob/5a6e01a5b1b6074081628658186812978fcdfadc/integration/admin-client-jee/src/main/java/org/keycloak/admin/client/resource/UserResource.java#L310

**Comment by Ketec** [comment]
25.0.2, I'm sending it from postman to test.

Also, in the admin UI,bit works if I link manually, I compared the requests, same payload, headers etc. Even tried with the same token as the admin UI requests used.

**Comment by Ketec** [comment, marked as answer]
Looks like i missed the mode raw for postman script and body was not in a acceptable format.
